HW3/P5/label_regions.cl

```diff
@@ -81,6 +81,49 @@ propagate_labels(__global __read_write int *labels,
 
     // CODE FOR PARTS 2 and 4 HERE (part 4 will replace part 2)
     
+    //PART 4
+    barrier(CLK_LOCAL_MEM_FENCE);
+
+
+    //use only one thread
+    if (lx == 0 && ly == 0)
+    {
+        for (int y_iterator = halo; y_iterator < buf_h - halo; y_iterator++)
+        {
+            for (int x_iterator = halo; x_iterator < buf_w - halo; x_iterator++)
+           {
+
+                int temp = buffer[y_iterator * buf_w + x_iterator];
```

Comment:
Your implementation of part 4 is wrong.
It is the same as part 2 (but with a single thread).